### PR TITLE
Fix card+status

### PR DESCRIPTION
**Pull request by vincit-matu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Fixes the remaining problems from TILA-3608, listed in the test plan below

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- `StatusLabel` colours should now be as intended [figma](https://www.figma.com/design/s987FF5VRRUNhDBvZJFPs9/HEL-Varaamo?node-id=15-85321&t=euEezHzE2c8rcAc5-1)
- The image should now fill intended space in horizontal cards, no matter how many lines of text it contains
- The default (mobile) cards are lined up as they should (proper paddings for the text, tag lines up with header)
- the body text in application round cards is now aligned to the bottom
- the arrow icon in upcoming application rounds status tag has been changed from topLeft to topRight
- on admin pages single reservation page, the status is indicated by a `StatusLabel`
- on admin pages the application states and application section states are indicated by `StatusLabel`s
- `StatusLabel`s should not make the tables any bigger (added `slim` option)
- the wordings are corrected as per the requests (listed in ticket)

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- TILA-3608

### CODE REVIEW

File: apps/admin-ui/src/spa/reservations/ReservationsTable.tsx [hunk 1]
```diff
@@ -50,19 +51,19 @@ const getStatusLabelProps = (
...
+      return { type: "neutral", icon: <IconCross aria-hidden /> };
     case ReservationStateChoice.Confirmed:
-      return { type: "success", icon: <IconCheck ariaHidden /> };
+      return { type: "success", icon: <IconCheck aria-hidden /> };
     case ReservationStateChoice.RequiresHandling:
-      return { type: "info", icon: <IconCogwheel ariaHidden /> };
+      return { type: "info", icon: <IconCogwheel aria-hidden /> };
     default:
-      return { type: "info", icon: <IconQuestionCircleFill ariaHidden /> };
+      return { type: "info", icon: <IconQuestionCircleFill aria-hidden /> };
```
Comment (joonatank) [inline apps/admin-ui/src/spa/reservations/ReservationsTable.tsx:66]: have to be `aria-hidden="true"` default value is not the same.

File: apps/admin-ui/src/spa/reservations/[id]/ReservationTitleSection.tsx [hunk 2]
```diff
@@ -134,6 +126,27 @@ const ReservationTitleSection = forwardRef<HTMLDivElement, Props>(
...
+        case ReservationStateChoice.Confirmed:
+          return { type: "success", icon: <IconCheck aria-hidden /> };
+        case ReservationStateChoice.Denied:
+          return { type: "error", icon: <IconCross aria-hidden /> };
+        case ReservationStateChoice.Cancelled:
+          return { type: "neutral", icon: <IconCross aria-hidden /> };
+        default:
+          return { type: "neutral", icon: <IconQuestionCircle aria-hidden /> };
+      }
+    })(reservation?.state);
```
Comment (joonatank) [inline apps/admin-ui/src/spa/reservations/[id]/ReservationTitleSection.tsx:148]: I'd make this into a free function (out of the component) not anonymous with an instant call.

Also `aria-hidden="true"`

File: apps/admin-ui/src/spa/reservations/[id]/ReservationTitleSection.tsx [hunk 3]
```diff
@@ -144,17 +157,20 @@ const ReservationTitleSection = forwardRef<HTMLDivElement, Props>(
                 <StatusLabel
                   type={statusLabelType}
                   data-testid="reservation_title_section__order_status"
-                  icon={<IconEuroSign ariaHidden />}
+                  icon={<IconEuroSign aria-hidden />}
```
Comment (joonatank) [inline apps/admin-ui/src/spa/reservations/[id]/ReservationTitleSection.tsx:160]: `aria-hidden="true"`

File: apps/ui/components/applications/ApplicationCard.tsx [hunk 4]
```diff
@@ -93,31 +93,31 @@ function getApplicationStatusLabelProps(
...
+        icon: <IconEnvelope aria-hidden />,
       };
     // These two should never be shown to the client, so they are shown as any other unexpected status
     case ApplicationStatusChoice.Cancelled:
     case ApplicationStatusChoice.Expired:
     default:
       return {
         type: "neutral",
-        icon: <IconQuestionCircle ariaHidden />,
+        icon: <IconQuestionCircle aria-hidden />,
```
Comment (joonatank) [inline apps/ui/components/applications/ApplicationCard.tsx:120]: `aria-hidden="true"`

File: packages/common/src/components/StatusLabel.tsx [hunk 5]
```diff
@@ -43,14 +47,16 @@ function StatusLabel({
...
   testId,
+  slim = false,
   children,
 }: Readonly<StatusLabelProps>): JSX.Element {
   return (
     <ColoredLabel
       type={type === "draft" ? "neutral" : type} // HDS StatusLabel does not support "draft" type - so convert it to "neutral"
       iconLeft={icon}
-      dataTestId={testId}
+      data-testId={testId}
```
Comment (joonatank) [inline packages/common/src/components/StatusLabel.tsx:57]: lower case `testid`

File: apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationsTable.tsx [hunk 6]
```diff
@@ -94,7 +118,15 @@ const COLS = [
     headerTKey: "Application.headings.phase",
     isSortable: true,
     key: "application_status",
-    transform: ({ statusView }: ApplicationView) => statusView,
+    transform: ({ status }: { status: ApplicationStatusChoice }) => {
+      const { t } = i18n;
```
Comment (joonatank) [inline apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationsTable.tsx:122]: use a hook `useTranslation` for this, not the `i18n` module directly.

File: apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationsTable.tsx [hunk 7]
```diff
@@ -38,8 +45,25 @@ type ApplicationView = {
...
+      return { type: "alert", icon: <IconClock aria-hidden /> };
+    case ApplicationStatusChoice.Received:
+      return { type: "info", icon: <IconCogwheel aria-hidden /> };
+    case ApplicationStatusChoice.Handled:
+      return { type: "success", icon: <IconCheck aria-hidden /> };
+    case ApplicationStatusChoice.ResultsSent:
+      return { type: "success", icon: <IconEnvelope aria-hidden /> };
+    default:
+      return { type: "neutral", icon: <IconQuestionCircle aria-hidden /> };
+  }
```
Comment (joonatank) [inline apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationsTable.tsx:66]: `aria-hidden="true"`

File: apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationEventsTable.tsx [hunk 8]
```diff
@@ -127,7 +156,15 @@ const COLS = [
     headerTKey: "ApplicationEvent.headings.phase",
     isSortable: true,
     key: "status",
-    transform: ({ statusView }: ApplicationEventView) => statusView,
+    transform: ({ status }: { status: ApplicationSectionStatusChoice }) => {
+      const { t } = i18n;
```
Comment (joonatank) [inline apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationEventsTable.tsx:160]: use the `useTranslation` hook

File: apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationEventsTable.tsx [hunk 9]
```diff
@@ -68,10 +80,27 @@ function appEventMapper(appEvent: Node): ApplicationEventView {
...
+    case ApplicationSectionStatusChoice.Unallocated:
+      return { type: "alert", icon: <IconArrowBottomRight aria-hidden /> };
+    case ApplicationSectionStatusChoice.InAllocation:
+      return { type: "info", icon: <IconCogwheel aria-hidden /> };
+    case ApplicationSectionStatusChoice.Handled:
+      return { type: "success", icon: <IconCheck aria-hidden /> };
+    case ApplicationSectionStatusChoice.Rejected:
+      return { type: "error", icon: <IconCross aria-hidden /> };
+    default:
+      return { type: "neutral", icon: <IconQuestionCircle aria-hidden /> };
```
Comment (joonatank) [inline apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationEventsTable.tsx:100]: `aria-hidden="true"`

**Review by joonatank — CHANGES_REQUESTED** [review]

File: apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationsTable.tsx (hunk 6 above)
Comment (vincit-matu) [reply apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationsTable.tsx]: Can't be used here, as `getColConfig` isn't inside a component.. but I refactored it so we don't have a `COLS` anymore, as discussed.

File: apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationEventsTable.tsx (hunk 8 above)
Comment (vincit-matu) [reply apps/admin-ui/src/spa/application-rounds/[id]/review/ApplicationEventsTable.tsx]: Can't be used here, as `getColConfig` isn't inside a component.. but I refactored it so we don't have a `COLS` anymore, as discussed.

**Review by joonatank — APPROVED** [review]

**Pull request merged**